test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/SelectExpandParserTests.cs

```diff
@@ -375,6 +375,68 @@ private SelectToken ParseSelectClause(string select)
         }
         #endregion
 
+        #region ParseSelectExpand
+        [Fact]
+        public void NoDollarNestedOptions()
+        {
+            // Act
+            SelectExpandParserTests.ParseSelectExpand(
+                select: "foo,bar(select=baz)",

```

Comment:
Look at other test cases in this file and use property names similar to those.
`foo` and `bar` in both Select and Expand is confusing.
**I understand we dont validate entities and properties in syntactic parsing, but less confusing property and entity names are better**

We can have something like
`select=RelatedSSNs,Addresses(select=City)`
`expand=Trips,Orders(select=OrderDetails; expand=Discounts)`